examples/acados_python/pendulum_on_cart/ocp/ocp_example_cost_formulations.py

```diff
@@ -199,6 +200,43 @@ def formulate_ocp(cost_version: str) -> AcadosOcp:
         ocp.model.cost_y_expr = ca.vertcat(x, u)
         ocp.model.cost_y_expr_e = x
         ocp.translate_cost_to_external_cost(cost_hessian='GAUSS_NEWTON')
+    elif cost_version == 'NLS_TO_EXTERNAL_P_GLOBAL':
+        ocp.cost.cost_type = 'NONLINEAR_LS'
+        ocp.cost.cost_type_e = 'NONLINEAR_LS'
+
+        p_global =struct_symSX([
+            entry('W', shape=(ny, ny)),
+            entry('yref', shape=(ny, )),
+            entry('W_e', shape=(ny_e, ny_e)),
+            entry('yref_e', shape=(ny_e, ))
+        ])
+        ocp.model.p_global = p_global.cat
+
+        ocp.cost.W = p_global['W']
+        ocp.cost.yref = p_global['yref']
+        ocp.cost.W_e = p_global['W_e']
+        ocp.cost.yref_e = p_global['yref_e']
+
+        ocp.model.cost_y_expr = ca.vertcat(x, u)
+        ocp.model.cost_y_expr_e = x
+
+        ocp.translate_cost_to_external_cost(cost_hessian='GAUSS_NEWTON')
+
+        p_global_values = p_global(0)
+        p_global_values['W'] = cost_W
+        p_global_values['yref'] = np.zeros((ny, ))
+        p_global_values['W_e'] = Q_mat
+        p_global_values['yref_e'] = np.zeros((ny_e, ))
+
+        ocp.p_global_values = p_global_values.cat.full().flatten()
+
+        ocp.cost.W_0 = np.zeros((0,0))
+        ocp.cost.W   = np.zeros((0,0))
+        ocp.cost.W_e = np.zeros((0,0))
+
+        ocp.cost.yref_0 = np.array([])
+        ocp.cost.yref   = np.array([])
+        ocp.cost.yref_e = np.array([])
```

Comment:
I would suggest removing those symbolic fields in `translate_cost_to_external_cost()`.
They can also be removed if they are not symbolic as `W`, `yref` etc are not part of the `EXTERNAL` cost formulation.